kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/net/EndpointListener.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.net;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import io.netty.handler.ssl.SslContext;
+
+import io.kroxylicious.proxy.config.TargetCluster;
+import io.kroxylicious.proxy.model.VirtualClusterModel;
+import io.kroxylicious.proxy.service.HostPort;
+
+/**
+ * A listener to an endpoint.
+ */
+public interface EndpointListener {
+    /**
+     * Target cluster associated with this listener.
+     * @return target cluster
+     */
+    TargetCluster targetCluster();
+
+    /**
+     * true if this listener uses TLS.
+     *
+     * @return true if listener uses TLS.
+     */
+    boolean isUseTls();
+
+    /**
+     * true if this listener requires TLS (in other words, use SNI).
+     * @return true if this listener requires TLS.
+     */
+    boolean requiresTls();
```

Comment:
Should the properties be `acceptsTls` and `requiresTls` ?

Reply:
looking at the usage in `EndpointRegistry` it looks like it's very SNI oriented:

```
var bindingKey = virtualCluster.requiresTls() ? RoutingKey.createBindingKey(host) : RoutingKey.NULL_ROUTING_KEY;
```
We use it to set up SNI related routing.

maybe it should be `requiresSni`, or if we want to try and make it more general `requiresHostnameForRouting` (it's pretty hard to hide this detail isn't it!)

Reply:
I'm +1 to renaming - `requiresSni` works for me.